src/renderers/dom/fiber/ReactDOMFiberEntry.js

```diff
@@ -172,15 +172,24 @@ var DOMRenderer = ReactFiberReconciler({
         namespace = root ? root.namespaceURI : getChildNamespace(null, '');
         break;
       }
-      default: {
-        const container: any = nodeType === COMMENT_NODE
-          ? rootContainerInstance.parentNode || rootContainerInstance
-          : rootContainerInstance;
+      case COMMENT_NODE: {
+        invariant(
+          rootContainerInstance.parentNode != null, // match null & undefined
+          'The commentNode which you are rendering into must have a parent.',
```

Comment:
Is this the best place for this invariant? I'm curious if there's any way we can validate earlier. Or do we have this problem because early validation didn't help?

Reply:
We could probably check earlier - It's uncertain where in the process the `parentNode` is removed, so I'd rather leave it here until we know that.

Reply:
How does adding an invariant in the same place help us? Presumably this trades one crash for another?

Reply:
The only benefit is making the error more clear - in hopes that
A) someone else might see it and realize their code is causing this situation.
B) If we ran into it again it would be clear faster what is going on.

If we knew what situation caused this, we could add a link to helpful info, although so far it seems like an unfortunately high-visibility corner case.

I think it would be nice to add a check for a parent if it's a commentNode in `isValidContainer`, which we use in several places to validate the container node. But again, not sure that would catch this since we don't know when the parentNode is getting unmounted.